### ISSUE TITLE
Warning: Failed prop type: Invalid prop `data` of type `object` supplied to `Carousel`, expected `array`.

### ISSUE THREAD

**Issue by jsappme** [archriss/react-native-snap-carousel#163]
Even so I use the following code:

```
<Carousel
data={ this.images instanceof Array ? this.images : [] }
... />
```

**Comment by bd-arc** [comment]
Can you share a typical value of `this.images`?

**Comment by jsappme** [comment]
["https://imageurl...", "https://xxx.."]

**Comment by bd-arc** [comment]
@jsappme I took the time to test [the provided](https://github.com/archriss/react-native-snap-carousel/tree/master/example) example with your the exact values you provided and didn't have any issue.

It must be something on your end, probably the way you handle images' fetching...

**Comment by jsappme** [comment]
Not sure why this issue is closed, I still have the issue:

```
[exp] Warning: PropTypes has been moved to a separate package. Accessing React.PropTypes is no longer supported and will be removed completely
in React 16. Use the prop-types package on npm instead. (https://fb.me/migrating-from-react-proptypes)
- node_modules\expo\src\Logs.js:187:24 in newConsoleFunc
- node_modules\fbjs\lib\warning.js:35:20 in printWarning
- node_modules\fbjs\lib\warning.js:59:25 in warning
- node_modules\react\cjs\react.development.js:2133:18 in get
* components\DropdownAlertProvider.js:10:20 in <unknown>
- node_modules\metro-bundler\src\Resolver\polyfills\require.js:174:12 in loadModuleImplementation
* App.js:3:0 in <unknown>
- node_modules\metro-bundler\src\Resolver\polyfills\require.js:174:12 in loadModuleImplementation
- node_modules\expo\AppEntry.js:2:0 in <unknown>
- ... 3 more stack frames from framework internals
```
Here is my code:

```
<Carousel
                    data={ this.props.property.images instanceof Array ? this.props.property.images : [] }
                    renderItem={ this.renderItem }
                    sliderWidth={width}
                    itemWidth={width}
                    inactiveSlideScale={1}
                    inactiveSlideOpacity={1}
                    enableMomentum={true}
                    activeSlideAlignment={'start'}
                    apparitionDelay={1500}
                    autoplay={true}
                    autoplayDelay={1900}
                    autoplayInterval={2500}
                    containerCustomStyle={{}}
                    contentContainerCustomStyle={{}}
                    removeClippedSubviews={false}
                    loop={false}
                  />
```

**Comment by bd-arc** [comment]
The issue has been closed since I wasn't able to reproduce it based solely on the information you provided.

The error you're currently facing has nothing to do with the original issue: it's a warning from React before version 16 (see [this post](https://reactjs.org/docs/typechecking-with-proptypes.html)). This has been fixed a long time ago, [in version `2.4.0`](https://github.com/archriss/react-native-snap-carousel/commit/9b07812647774499307b4bbbc397127e485fadb9). I wonder which version of the plugin you're using, because if it's anything before `3.0.0`, you need to take a look at the older documentation.

**Comment by schumannd** [comment]
I am experiencing this error now in conjunction with mobx ObservableArray.
